pkg/ibmvpc/sg_analysis_test.go

```diff
@@ -107,3 +110,101 @@ func TestSGRule(t *testing.T) {
+
+	c1 := connection.TCPorUDPConnection(netp.ProtocolString("UDP"), 5, 87, 10, 3245)
+	c2 := connection.TCPorUDPConnection(netp.ProtocolString("TCP"), 1, 100, 5, 1000)
+
+	rulesTest1 := []*SGRule{
+		{
+			target:      cidr1t,
+			connections: c1,
+			index:       1,
+			local:       ipblock.GetCidrAll(),
+		},
+		{
+			target:      cidr2t,
+			connections: c2,
+			index:       2,
+			local:       ipblock.GetCidrAll(),
+		},
+	}
+
+	target1, _ := ipblock.FromCidrOrAddress("10.250.10.0")
+	target2, _ := ipblock.FromCidrOrAddress("10.250.10.2")
+	target3, _ := ipblock.FromCidrOrAddress("10.250.10.1")
+
+	egressConnectivityMap := make(map[*ipblock.IPBlock]*ConnectivityResult)
+	mapAndAnalyzeSGRules(rulesTest1, false, egressConnectivityMap)
+	require.Equal(t, len(egressConnectivityMap), 1)
+	for _, CR := range egressConnectivityMap {
+		for remote, conn := range CR.allowedConns {
+			if target1.ContainedIn(remote) || target2.ContainedIn(remote) {
+				require.True(t, conn.Equal(c2))
+			}
+			if target3.ContainedIn(remote) {
+				require.True(t, c1.ContainedIn(conn) && c2.ContainedIn(conn))
+			}
+		}
+	}
+}
+
+func TestLocalField(t *testing.T) {
+	cidr1, _ := ipblock.FromCidrOrAddress("10.240.10.1")
+	cidr2, _ := ipblock.FromCidrOrAddress("10.240.10.0/30")
+	cidr1t, _ := ipblock.FromCidrOrAddress("10.250.10.1")
+	cidr2t, _ := ipblock.FromCidrOrAddress("10.250.10.0/30")
+
+	c1 := connection.TCPorUDPConnection(netp.ProtocolString("UDP"), 5, 87, 10, 3245)
+	c2 := connection.TCPorUDPConnection(netp.ProtocolString("TCP"), 1, 100, 5, 1000)
+
+	rulesTest1 := []*SGRule{
+		{
+			target:      cidr1t,
+			connections: c1,
+			index:       1,
+			local:       cidr1,
+		},
+		{
+			target:      cidr2t,
+			connections: c2,
+			index:       2,
+			local:       cidr2,
+		},
+	}
+
+	member1, _ := ipblock.FromCidrOrAddress("10.240.10.0")
+	member2, _ := ipblock.FromCidrOrAddress("10.240.10.2")
+	member3, _ := ipblock.FromCidrOrAddress("10.240.10.1")
+
+	target1, _ := ipblock.FromCidrOrAddress("10.250.10.0")
+	target2, _ := ipblock.FromCidrOrAddress("10.250.10.2")
+	target3, _ := ipblock.FromCidrOrAddress("10.250.10.1")
+
+	egressConnectivityMap := make(map[*ipblock.IPBlock]*ConnectivityResult)
+	mapAndAnalyzeSGRules(rulesTest1, false, egressConnectivityMap)
+	for local, connectivityResult := range egressConnectivityMap {
+		for remote, conn := range connectivityResult.allowedConns {
+			if !local.ContainedIn(cidr2) || !remote.ContainedIn(cidr2t) {
+				// no connection
+				continue
+			}
+			switch {
+			case member1.ContainedIn(local):
+				require.True(t, conn.Equal(c2))
+			case member2.ContainedIn(local):
+				require.True(t, conn.Equal(c2))
+			case member3.ContainedIn(local):
+				if target1.ContainedIn(remote) || target2.ContainedIn(remote) {
+					require.True(t, conn.Equal(c2))
+				}
+				if target3.ContainedIn(remote) {
+					require.True(t, c1.ContainedIn(conn) && c2.ContainedIn(conn))
+				}
+			}
+		}
+	}
```

Comment:
same comment here.. unit tests should better be explicit, rather than code from which it is difficult to understand what properties are checked and why.